runtime/runtime-params-estimator/src/cost.rs

```diff
@@ -60,15 +83,81 @@ pub enum Cost {
+    /// `Action::Stake`, a slightly higher value for sending than executing.
+    ///
+    /// Estimation: Measure a transaction with only a staking action and
+    /// subtract the base cost of creating a sir-receipt.
+    /// (TODO[jakmeier] find out and document the reasoning behind send vs exec
+    /// values in this specific case)
     ActionStake,
+    /// Estimates `action_creation_config.add_key_cost.full_access_cost` which
+    /// is charged for every `Action::AddKey` where the key is a full access
+    /// key. The same value is charged for sending and executing.
+    ///
+    /// Estimation: Measure a transaction that adds a full access key and
+    /// subtract the base cost of creating a sir-receipt.
     ActionAddFullAccessKey,
+    /// Estimates `action_creation_config.add_key_cost.function_call_cost` which
+    /// is charged once for every `Action::AddKey` where the key is a function
+    /// call key. The same value is charged for sending and executing.
+    ///
+    /// Estimation: Measure a transaction that adds a function call key and
+    /// subtract the base cost of creating a sir-receipt.
     ActionAddFunctionAccessKeyBase,
+    /// Estimates
+    /// `action_creation_config.add_key_cost.function_call_cost_per_byte` which
+    /// is charged once for every byte in null-terminated method names listed in
+    /// an `Action::AddKey` where the key is a function call key. The same value
+    /// is charged for sending and executing.
+    ///
+    /// Estimation: Measure a transaction that adds a function call key with
+    /// many methods. Subtract the cost of adding a function call key with a
+    /// single method and of creating a sir-receipt. The result is divided by
+    /// total bytes in the method names.
     ActionAddFunctionAccessKeyPerByte,
+    /// Estimates `action_creation_config.delete_key_cost` which is charged for
+    /// `DeleteKey` actions, the same value on sending and executing. It does
+    /// not matter whether it is a function call or full access key.
+    ///
+    /// Estimation: Measure a transaction that deletes a full access key and
+    /// transfers an initial balance to it. Subtract the base cost of creating a
+    /// receipt.
+    /// (TODO[jakmeier] check cost for function call keys with many methods)
     ActionDeleteKey,
+    /// Estimates `action_creation_config.delete_account_cost` which is charged
+    /// for `DeleteAccount` actions, the same value on sending and executing.
+    ///
+    /// Estimation: Measure a transaction that deletes an existing account.
+    /// Subtract the base cost of creating a sir-receipt.
+    /// (TODO[jakmeier] Consider different account states.
     ActionDeleteAccount,
 
+    /// Estimates `wasm_config.ext_costs.base` which is intended to be charged
+    /// once on every host function call. However, this is currently
+    /// inconsistent. First, we do not charge on calls to `sha256`. Furthermore,
```

Comment:
Is the `sha256` the only exception? I didn't check that :)

Reply:
Hehe, silly me, should have checked. It is actually the same for all math API functions. Which sort of makes sense, each already has it's own base cost, so it might as well be included there.

Otherwise, it seems rather consistent. Deprecated host functions don't have it but I don't think this needs mentioning.